File: services/socket-service/test_socket.js
const axios = require('axios');

// URLs
const API_GATEWAY_URL = 'http://localhost:8081';
const SOCKET_SERVICE_DIRECT_URL = 'http://localhost:3002';

// Test the socket service health endpoint via API Gateway
const testSocketService = async () => {
  try {
    console.log('Testing Socket Service...');

    // Try direct access to socket service health endpoint via API Gateway
    console.log('\nTesting API Gateway proxy to Socket service:');
    try {
      const response = await axios.get(`${API_GATEWAY_URL}/socket-http/health`);
      console.log('✅ Socket service health via API Gateway:', response.data);
    } catch (error) {
      console.error('❌ Error accessing socket service via API Gateway:', error.message);
    }

    console.log('\nTest complete.');
  } catch (error) {
    console.error('Test failed with error:', error.message);
  }
};

// Run the test
testSocketService(); 

File: services/socket-service/test_socket.py
#!/usr/bin/env python3
import requests
import json
import sys

def test_socket_service():
    """Test the socket service API endpoints"""
    api_gateway_url = "http://localhost:8081"
    
    print("Testing Socket Service API...")
    
    # Test health endpoint
    print("\n1. Testing health endpoint:")
    try:
        response = requests.get(f"{api_gateway_url}/socket-http/health")
        print(f"✅ Health check: {response.status_code} - {response.text}")
    except Exception as e:
        print(f"❌ Health check failed: {e}")
    
    # Test start-execution endpoint without authentication (expect failure)
    print("\n2. Testing start-execution endpoint (without auth - should fail):")
    try:
        execution_data = {
            "executionId": "test-execution-123",
            "agent": "test-agent",
            "input": "test command",
            "userId": "test-user-123"
        }
        response = requests.post(
            f"{api_gateway_url}/socket-http/api/start-execution", 
            json=execution_data
        )
        print(f"Response: {response.status_code} - {response.text}")
    except Exception as e:
        print(f"❌ Start execution test failed: {e}")
    
    # Test execution-status endpoint
    print("\n3. Testing execution-status endpoint:")
    try:
        status_data = {
            "executionId": "test-execution-123",
            "status": "running",
            "progress": 50,
            "output": "Test output"
        }
        response = requests.post(
            f"{api_gateway_url}/socket-http/api/execution-status", 
            json=status_data
        )
        print(f"Response: {response.status_code} - {response.text}")
    except Exception as e:
        print(f"❌ Execution status test failed: {e}")
    
    print("\nTests completed.")

if __name__ == "__main__":
    test_socket_service() 